src/Microplate.php

```diff
@@ -0,0 +1,187 @@
+<?php declare(strict_types=1);
+
+namespace Mll\Microplate;
+
+use Illuminate\Support\Collection;
+use Mll\Microplate\Enums\FlowDirection;
+use Mll\Microplate\Exceptions\MicroplateIsFullException;
+use Mll\Microplate\Exceptions\UnexpectedFlowDirection;
+use Mll\Microplate\Exceptions\WellNotEmptyException;
+
+/**
+ * @template TWell
+ * @template TCoordinateSystem of CoordinateSystem
+ *
+ * @phpstan-type WellsCollection Collection<array{Coordinate<TCoordinateSystem>, TWell}>
+ */
+class Microplate
+{
+    public const EMPTY_WELL = null;
+
+    /**
+     * @var TCoordinateSystem
+     */
+    public CoordinateSystem $coordinateSystem;
+
+    /**
+     * @var WellsCollection
+     */
+    public Collection $wells;
+
+    /**
+     * @param TCoordinateSystem $coordinateSystem
+     */
+    public function __construct(CoordinateSystem $coordinateSystem)
+    {
+        $this->coordinateSystem = $coordinateSystem;
+
+        $this->clearWells();
+    }
+
+    /**
+     * @param Coordinate<TCoordinateSystem> $coordinate
+     * @param TWell $content
+     *
+     * @throws WellNotEmptyException
+     */
+    public function addWell(Coordinate $coordinate, $content): void
+    {
+        $this->assertIsWellEmpty($coordinate, $content);
+        $this->setWell($coordinate, $content);
+    }
+
+    /**
+     * @param TWell $content
+     *
+     * @throws MicroplateIsFullException
+     */
+    public function addToNextFreeWell($content, FlowDirection $flowDirection): void
```

Comment:
@spawnia what do you think about returning the Coordinate from this method?
I reckon that one wants to save the coordinate e.g. to the database after adding a sample...

Anything that speaks against this suggestion?

Reply:
Sounds fine